pyuvdata/uvbeam/cst_beam.py

```diff
@@ -220,36 +226,127 @@ def read_cst_beam(
             raise ValueError(
-                "Data does not appear to be regularly gridded in zenith angle"
+                "The zenith-angle is outside (0, pi): "
+                f"{theta_data.min(), theta_data.max()}. "
+                "Check your thetaphi_to_azza_fnc."
             )
 
-        if not uvutils._test_array_constant_spacing(phi_axis, self._axis1_array.tols):
+        if phi_data.max() > phi_data.min() + 2 * np.pi:
             raise ValueError(
-                "Data does not appear to be regularly gridded in azimuth angle"
+                "The azimuthal angle is outside (0, 2*pi): "
+                f"{phi_data.min(), phi_data.max()}. "
+                "Check your thetaphi_to_azza_fnc."
             )
+
+        phi_data %= 2 * np.pi
+
+        theta_axis = np.sort(np.unique(theta_data))
+        phi_axis = np.sort(np.unique(phi_data))
+
+        # Get the order of the data in which phi increases on the inner loop and theta
+        # on the outer.
+        sort_idx = np.lexsort((phi_data, theta_data))
+        theta_data = theta_data[sort_idx]
+        phi_data = phi_data[sort_idx]
+        data = data[sort_idx]
+        data[:, theta_col] = theta_data
+        data[:, phi_col] = phi_data
+
+        if self.beam_type == "power":
+            data_col_enum = ["abs(e)", "abs(v)"]
+            data_col = []
+            for name in data_col_enum:
+                this_col = np.where(np.array(column_names) == name)[0]
+                if this_col.size > 0:
+                    data_col = data_col + this_col.tolist()
+            if len(data_col) == 0:
+                raise ValueError(f"No power column found in file: {filename}")
+            elif len(data_col) > 1:
+                raise ValueError(
+                    f"Multiple possible power columns found in file: {filename}"
+                )
+            data_col = data_col[0]
+
+        tupled = list(zip(theta_data, phi_data))
+        n = len(tupled)
+        new_data = np.zeros((len(theta_axis) * len(phi_axis), data.shape[1]))
+        i = 0
+        idx = -1
+        for za in theta_axis:
+            for az in phi_axis:
+                if (idx + 1) < n and (za, az) == tupled[idx + 1]:
+                    # best case scenario is everything is in grid already.
+                    # For most beams, the large majority of iterations will hit this.
+                    idx = idx + 1
+                    new_data[i] = data[idx]
+                else:
+                    try:
+                        # Find the index of these coordinates
+                        # There may be multiple indices with the same coords, because
+                        # of poles -- here we just take the first one.
+                        idx = tupled.index((za, az))
+                        new_data[i] = data[idx]
+                        if np.isclose(new_data[i, theta_col], -np.pi):
+                            print(data[idx], za, az)
```

Comment:
What is this print for?

Reply:
woops, removed

Reply:
I don't think you committed this change.